File: Ejercicio 6/Ejercicio 6.js
function main() {
    let array = new Array(15);
    
    // Generar números aleatorios
    for (let i = 0; i < array.length; i++) {
        array[i] = Math.floor(Math.random() * 100) + 1;
    }
    
    console.log("Array original:");
    mostrarArray(array);
    
    // Ordenamiento por inserción
    for (let i = 1; i < array.length; i++) {
        let elementoActual = array[i];
        let j = i - 1;
        
        while (j >= 0 && array[j] > elementoActual) {
            array[j + 1] = array[j];
            j--;
        }
        array[j + 1] = elementoActual;
    }
    
    console.log("Array ordenado:");
    mostrarArray(array);
}

function mostrarArray(array) {
    let resultado = "";
    for (let num of array) {
        resultado += num + " ";
    }
    console.log(resultado);
}

// Ejecutar el programa
main();

File: Ejercicio 6/Ejercicio 6.py
import random

def mostrar_array(array):
    for num in array:
        print(num, end=" ")
    print()

def main():
    array = [0] * 15
    
    # Generar números aleatorios
    for i in range(len(array)):
        array[i] = random.randint(1, 100)
    
    print("Array original:")
    mostrar_array(array)
    
    # Ordenamiento por inserción
    for i in range(1, len(array)):
        elemento_actual = array[i]
        j = i - 1
        
        while j >= 0 and array[j] > elemento_actual:
            array[j + 1] = array[j]
            j -= 1
        
        array[j + 1] = elemento_actual
    
    print("Array ordenado:")
    mostrar_array(array)

# Ejecutar el programa
if __name__ == "__main__":
    main()